[Detection Engine] Rule Data View select uses Data View name for display

rylnd opened the pull request:
## Summary

Supersedes #190936, and also addresses https://github.com/elastic/kibana/issues/137823. This mainly just rebases the changes introduced there to the latest `main`. I also noticed that it originally had some unit tests, so I resurrected those as well. ~~I modified the copy as well, so nota bene @ARWNightingale~~ After some discussion, we determined that an explicit UI note about this behavior was unnecessary.


## Screenshots

<kbd>
<img width="979" alt="Screenshot 2025-03-13 at 5 24 42 PM" src="https://github.com/user-attachments/assets/69da29bf-5504-461e-b797-d016adcd4b59" /></kbd>

<kbd><img width="1052" alt="Screenshot 2025-03-18 at 4 37 58 PM" src="https://github.com/user-attachments/assets/8ee5e55e-2413-4936-aa87-7dadb3d7cbb1" /></kbd>




### Checklist


- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios



<!--ONMERGE {"backportTargets":["7.17","8.16","8.17","8.18","8.x","9.0"]} ONMERGE-->

yctercero committed:
updating to use new component

This ports Yara's code from #190936 to a new branch based on the latest
upstream.

 Conflicts:
	x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/schema.tsx

rylnd committed:
Another pass on copy

rylnd committed:
Add a unit test validating that we display options by name

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/283931)
* Commit: 70031c816d7f6b5851fea1bf0f1ac15ca974e180

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d66e0b25258b0568857bf6a0882e4d40e1f7e771) | [after](https://github.com/elastic/kibana/commit/70031c816d7f6b5851fea1bf0f1ac15ca974e180) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | +257.0B |



cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283931","number":283931,"commit":"70031c816d7f6b5851fea1bf0f1ac15ca974e180"}],"number":283931}
buildkite-pr-comment-->

rylnd committed:
Merge branch 'main' into rylnd/data_view_select_uses_name_for_display

rylnd commented:
@nastasha-solomon pinging you here for review on the copy; I don't think @ARWNightingale's review is needed now, as only copy has changed.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/284795)
* Commit: e238645343ae56c8ffbdb1d196e671795c90c767

### Failed CI Steps
* [Defend Workflows Cypress Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/284795#0195a632-72d8-438d-a61d-36b6ff58dc56)
* [Serverless Detection Engine - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/284795#0195a632-72e1-4afd-8963-283e7a6c8515)

### History
* :green_heart: [Build #283931](https://buildkite.com/elastic/kibana-pull-request/builds/283931) succeeded 70031c816d7f6b5851fea1bf0f1ac15ca974e180

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284795","number":284795,"commit":"e238645343ae56c8ffbdb1d196e671795c90c767"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283931","number":283931,"commit":"70031c816d7f6b5851fea1bf0f1ac15ca974e180"}],"number":284795}
buildkite-pr-comment-->

nastasha-solomon commented on x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/schema.tsx:
```diff
@@ -68,6 +68,13 @@ export const schema: FormSchema<DefineStepRule> = {
+      'xpack.securitySolution.detectionEngine.createRule.stepAboutRule.dataViewSelector.helpText',
+      {
+        defaultMessage:
+          'Data views are displayed by name. To see their full definitions, visit Stack Management > Data Views.',
```
Is there any way to link to the Data Views page in ESS and Serverless? The paths to the page are slightly different:
- **ESS**: Management > Stack Management > Kibana > Data Views 
- **Serverless**: Project Settings > Management > Data > Data Views

rylnd replied:
@nastasha-solomon given that we link to the Data Views documentation at the top of the component:

<kbd><img width="1052" alt="Screenshot 2025-03-18 at 2 30 59 PM" src="https://github.com/user-attachments/assets/f630afb0-4086-46f4-bf49-b0f4b29a5a7a" /></kbd>

Perhaps we should just state the fact? In Discover, for example, "display by name" is the default with no explanation, and the fact of this issue/PR sort of implies that this is the de facto representation moving forward. Should we simply remove this help text entirely? We have opportunity to note the change in release notes, for any users that might be confused.

yctercero replied:
I think it makes sense to move closer towards a unified UX - sourcerer is also moving towards utilizing the same component as Discover. 

I think the link to data view or index patterns also links to the wrong page. I think I remember there being a ticket about that.

nastasha-solomon replied:
Yeah, that's a good point. If Discover changes the way it displays data view options (for example, it reverts back to displaying the definition instead of the name), we'd likely need to change the text again. I'm good with removing the help text entirely.

nastasha-solomon replied:
@yctercero looks like it currently links to [these](https://www.elastic.co/guide/en/kibana/8.18/data-views.html) Kibana docs. Would you prefer that it link to [this](https://www.elastic.co/guide/en/security/current/data-views-in-sec.html) Security page insted?

rylnd replied:
I removed the help text in https://github.com/elastic/kibana/pull/214495/commits/912e75aa901652c904b495785e125eac8a70dd3d. Please approve/deny based on your agreement.

yctercero replied:
Looks good, thanks @rylnd !

rylnd committed:
Merge branch 'main' into rylnd/data_view_select_uses_name_for_display

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285169)
* Commit: 0c84d6fb318e975bdd0fa41bf43578c52453ad66

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ee906e67e2127d141ef2b9e255adca58615038de) | [after](https://github.com/elastic/kibana/commit/0c84d6fb318e975bdd0fa41bf43578c52453ad66) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | +257.0B |



### History
* :broken_heart: [Build #284795](https://buildkite.com/elastic/kibana-pull-request/builds/284795) failed e238645343ae56c8ffbdb1d196e671795c90c767
* :green_heart: [Build #283931](https://buildkite.com/elastic/kibana-pull-request/builds/283931) succeeded 70031c816d7f6b5851fea1bf0f1ac15ca974e180

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285169","number":285169,"commit":"0c84d6fb318e975bdd0fa41bf43578c52453ad66"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284795","number":284795,"commit":"e238645343ae56c8ffbdb1d196e671795c90c767"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283931","number":283931,"commit":"70031c816d7f6b5851fea1bf0f1ac15ca974e180"}],"number":285169}
buildkite-pr-comment-->

rylnd committed:
Remove help text from DataView selector on rule creation

Since the default (and expected, as per this PR) behavior is to display
Data Views by name, additional UI text indicating this behavior is noisy
and arguably unnecessary.

Instead, we will have the release not accompanying this enhancement to
assuage any confused users.

nastasha-solomon reviewed (approved):
👍🏼 to removing the help text.

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/285193)
* Commit: 912e75aa901652c904b495785e125eac8a70dd3d

### Failed CI Steps
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/285193#0195ab53-3244-4712-aecf-e5f794cc53a0)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/285193#0195ab53-3244-4712-aecf-e5f794cc53a0) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/285193/jobs/0195ab53-3244-4712-aecf-e5f794cc53a0/artifacts/0195ab73-02ef-425f-bb43-2b1975e8d740) FTR Configs #<span></span>13 / Package policies Package Policy - upgrade when package version is not installed "after each" hook for "should respond with an error"

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ee906e67e2127d141ef2b9e255adca58615038de) | [after](https://github.com/elastic/kibana/commit/912e75aa901652c904b495785e125eac8a70dd3d) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8.9MB | 8.9MB | +17.0B |



### History
* :green_heart: [Build #285169](https://buildkite.com/elastic/kibana-pull-request/builds/285169) succeeded 0c84d6fb318e975bdd0fa41bf43578c52453ad66
* :broken_heart: [Build #284795](https://buildkite.com/elastic/kibana-pull-request/builds/284795) failed e238645343ae56c8ffbdb1d196e671795c90c767
* :green_heart: [Build #283931](https://buildkite.com/elastic/kibana-pull-request/builds/283931) succeeded 70031c816d7f6b5851fea1bf0f1ac15ca974e180

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285193","number":285193,"commit":"912e75aa901652c904b495785e125eac8a70dd3d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/285169","number":285169,"commit":"0c84d6fb318e975bdd0fa41bf43578c52453ad66"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/284795","number":284795,"commit":"e238645343ae56c8ffbdb1d196e671795c90c767"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/283931","number":283931,"commit":"70031c816d7f6b5851fea1bf0f1ac15ca974e180"}],"number":285193}
buildkite-pr-comment-->

kibanamachine commented:
## 💔 Some backports could not be created

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|7.17|Backport failed because of merge conflicts|
|❌|8.16|Backport failed because of merge conflicts|
|✅|8.17|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/215351">](https://github.com/elastic/kibana/pull/215351)|
|✅|8.18|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/215353">](https://github.com/elastic/kibana/pull/215353)|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/215354">](https://github.com/elastic/kibana/pull/215354)|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/215355">](https://github.com/elastic/kibana/pull/215355)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 214495
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

mistic commented:
This PR didn't land on time for the latest 8.17.4 BC. Updating the labels.

nicpenning commented:
Related : https://github.com/elastic/kibana/issues/168909
